test/Tests/src/Data/List_Spec.enso

```diff
@@ -38,6 +38,44 @@ spec = Test.group "List" <|
         l.contains 4 . should_be_false
         l.contains 3 . should_be_true
         empty.contains 10 . should_be_false
+    Test.specify "should allow finding an element in the list with `.find`" <|
+        l.find (==2) . should_equal 2
+        l.find (==3) . should_equal 3
+        l.find (==4) . should_fail_with Not_Found
+        l.find (==2) start=1 . should_equal 2
+        l.find (==2) start=2 . should_fail_with Not_Found
+        l.find (==2) start=-1 . should_fail_with Not_Found
+        l.find (==2) start=-2 . should_equal 2
+        empty.find (==1) . should_fail_with Not_Found
+        empty.find (==1) if_missing=Nothing . should_equal Nothing
+    Test.specify "should allow finding the index of an element in the list with `.index_of`" <|
+        l.index_of (==2) . should_equal 1
+        l.index_of 3 . should_equal 2
+        l.index_of (==4) . should_equal Nothing
+        empty.index_of (==1) . should_equal Nothing
+        l.index_of (==2) start=1 . should_equal 1
+        l.index_of 2 start=2 . should_equal Nothing
+        l.index_of 2 start=3 . should_equal Nothing
+        l.index_of 2 start=-1 . should_equal Nothing
+        l.index_of (==2) start=-2 . should_equal 1
+        l.index_of 2 start=4 . should_fail_with Index_Out_Of_Bounds.Error
```

Comment:
Given that the logic for computing the index/offset in the List functions is especially complex and it is super easy to do some off-by-one errors there, I'd really like to see some tests (ideally we should replace all `should_fail_with` here with that pattern) checking if the error is actually what we expect, for example:
```suggestion
        l.index_of 2 start=4 . catch . should_equal (Index_Out_Of_Bounds.Error 4 3)
```